project/MiMa.scala

```diff
@@ -0,0 +1,94 @@
+// It would be nice to use sbt-mima-plugin here, but the plugin is missing
+// at least two features we need:
+// * ability to run MiMa twice, swapping `curr` and `prev`, to detect
+//   both forwards and backwards incompatibilities
+// * ability to pass a filter file
+// So we invoke the MiMa CLI directly; it's also what the Ant build did.
+
+import sbt._
+import sbt.Keys._
+
+object MiMa {
+
+  lazy val mimaOldVersion =
+    settingKey[String]("Scala version number to run MiMa against")
+
+  lazy val mima =
+    taskKey[Unit]("run Migration Manager to detect binary incompatibilities")
+
+  lazy val settings =
+    Seq(
+      mimaOldVersion := VersionUtil.versionProps("scala.binary.version") + ".0",
```

Comment:
not sure if using `scala.binary.version` is correct here, see its doc comment and its value in the 2.12.x branch: https://github.com/scala/scala/blob/2.12.x/versions.properties#L16-L22

the code here will only work after `.0` is released. also note that we'd like to enable mima once M5 is out and set M5 (or maybe later RC1) as the base version. in ant we used `2.11.0` directly, no?

related, what is the scenario to enable / disable mima during the milestone cycle? just don't invoke it from the CI bash scripts?